src/uwtools/rocoto.py

```diff
@@ -244,6 +264,39 @@ def _add_workflow_tasks(self, e: Element, config: dict) -> None:
             tag, name = self._tag_name(key)
             {STR.metatask: self._add_metatask, STR.task: self._add_task}[tag](e, block, name)
 
+    def _config_validate(self, config_file: OptionalPath) -> None:
+        """
+        Validate the given YAML config.
+
+        :param config_file: Path to the YAML config (defaults to stdin).
+        """
+        if not validate_yaml(
+            config_file=config_file, schema_file=resource_pathobj("rocoto.jsonschema")
+        ):
+            raise UWConfigError("YAML validation errors identified in %s" % config_file)
+
+    @property
+    def _dependency_constants(self):
```

Comment:
I wouldn't mind seeing a `-> Tuple[Tuple, Tuple, Tuple]` annotation for the return type on the method signature. It looks a bit silly, but it's accurate.

Reply:
That brings up an excellent point. We are importing `Tuple` like so ```from typing import Optional, Tuple```. The return type on the signature I can update to ```-> Tuple[Tuple, Tuple, Tuple]``` accordingly. 

Previously when I was passing `dependency_constants` in as an argument I used `tuple` as the input type. I wanted to confirm that here the difference between `Tuple` vs. `tuple` is lower case `tuple` is being used a type in python. Whereas `Tuple` is imported from the typing module. I was trying to find more context to why we would prefer one over the other.

I'll update that return type now.

Reply:
You can use the built-in type names (`dict`, `list`, `tuple`, etc.) as types as long as you don't want to be more specific than that -- i.e. you're ok with "a dict of anything at all", "a list of anything at all", etc.

The ones imported from `typing` let you add square brackets and more precise information. So, for example, we could just say `-> tuple` or `-> Tuple`, which would be equivalent, but would mean that the tuples could contain anything; or `Tuple[tuple, tuple, tuple]` (lower-case, built-in versions this time) would be equivalent to what I suggested.

Or we could go even more specific: `Tuple[Tuple[str, str, str], Tuple[str, str, str, str, str, str], Tuple[str, str]]`. That would say everything there is to say, down to how many `str` elements each tuple should have, but seemed a bit much to me.

I think there's also a middle ground like `Tuple[Tuple[str, ...], Tuple[str, ...], Tuple[str, ...]]` which at least says that all the inner tuples must contain only strings, but we're not staying how many. For convenience, we might define a custom type, like `StrTuple = Tuple[str, ...]`, and then we could say `Tuple[StrTuple, StrTuple, StrTuple]`.

I'm not sure what's best.